AirCasting/BluetoothSession/UserTriggeredReconnectionController.swift

```diff
@@ -0,0 +1,67 @@
+// Created by Lunar on 24/11/2022.
+//
+
+import Foundation
+import Resolver
+
+enum UserTroggeredReconnectionError: Error {
+    case anotherActiveSessionInProgress
+    case failedToDiscoverCharacteristics
+    case failedToConnect
+    case deviceNotDiscovered
+    case airbeamConfigurationFailure
+}
+
+protocol UserTriggeredReconnectionController {
+    func reconnectWithPeripheral(deviceUUID: String, session: Session, completion: @escaping (Result<Void, UserTroggeredReconnectionError>) -> Void)
+}
+
+class DefaultUserTriggeredReconnectionController: UserTriggeredReconnectionController {
+    @Injected private var btScanner: BluetoothScanner
+    @Injected private var bluetootConnector: BluetoothConnectionHandler
+    @Injected private var activeSessionProvider: ActiveMobileSessionProvidingService
+    @Injected private var sessionRecorder: BluetoothSessionRecordingController
+    
+    func reconnectWithPeripheral(deviceUUID: String, session: Session, completion: @escaping (Result<Void, UserTroggeredReconnectionError>) -> Void) {
+        guard activeSessionProvider.activeSession == nil else {
+            Log.info("Tried to resume session when there was another active session in progress")
+            completion(.failure(UserTroggeredReconnectionError.anotherActiveSessionInProgress))
+            return
+        }
+        
+        var discoveredDevices: [NewBluetoothManager.BluetoothDevice] = []
+        btScanner.startScanning(scanningWindow: 10, onDeviceDiscovered: { discoveredDevices.append($0) }, onScanningFinished: {
```

Comment:
I feel like this can be significantly improved: why do we wait until the scanning is finished here? We can use `onDeviceDiscovered` to check if that's "our" device and do stuff there (stop scanning and reconnect). We will probably go down to 1-2 sec instead od 10. Or am I missing something here?

Reply:
Yeah, good catch, thanks

Reply:
I'm not able to implement it so I would leave it as is since this won't be used by many user and I don't want to spend too much time on it